### DISCUSSION TITLE
Issue with Map api path with double slash

### DISCUSSION THREAD

**Discussion by npalla** [dotnet/yarp#744] [Q&A] [answered]
We are trying to move from proxy-kit to reverse-proxy and and working fine, But we have issue with double slashes. If the api path has double slashes like http://dev.localdev.tcpci.com/app/managepermit//extapi/system/setting, the request is rejected as 404 Not Found inside Linux Container. If I need to update/cleanup the path to remove double slash, what's the best place to do the same. Any suggestions?

Thanks

**Comment by Tratcher** [comment, marked as answer]
That value is coming from the client and is not an assembly mistake by the proxy, right?

The fixup is a simple string replace, but deciding when to do it is interesting. From your prior post, it looks like you're using IHttpProxy directly? https://github.com/microsoft/reverse-proxy/discussions/736. In that case you'd want to fix the path either before calling ProxyAsync or inside the request transform callback. Here's what a request transform would look like:
https://github.com/microsoft/reverse-proxy/blob/cd117da030ca16d201de9d579e11155ae040d791/samples/ReverseProxy.Direct.Sample/Startup.cs#L64-L72

Here's how the request uri gets built (with some allocation improvements that just got made). You can modify the path while assembling the proxyRequest.RequestUri in your own transform.
https://github.com/microsoft/reverse-proxy/pull/740/files

**Reply by npalla** [reply]
Its not the proxy URL that's getting 404, its the UI request that comes with doubleslash 
![image](https://user-images.githubusercontent.com/3278465/108071357-a3e0e500-7033-11eb-93e9-d4a03ef1c7d8.png)

Also tried with postman .. with doubleslash will get 404, and without it returns data properly. I am trying to find a way in the middle ware to cleanup the URL before it starts to process the route. FYI - The proxy URL has no issues with double slash as its not hosted in container. Any suggestions?

**Reply by npalla** [reply]
if we use Map("/{**catch-all}") with a check to see if my route has "egapi" it works fine.  But I believe we hit the reverse proxy code for every api request from UI. I am just concerned if that has any performance hit for heavy traffic.

**Reply by npalla** [reply]
![image](https://user-images.githubusercontent.com/3278465/108073413-01763100-7036-11eb-8d79-a48565763db1.png)

![image](https://user-images.githubusercontent.com/3278465/108075341-40a58180-7038-11eb-862c-541455bd0847.png)

**Reply by Tratcher** [reply]
If it's messing with the proxy routing then you can fix the path before routing happens.
```C#
            app.Use((context, next) =>
            {
                var path = context.Request.Path.Value;
                path = path.Replace("//", "/");
                context.Request.Path = new PathString(path);

                return next();
            });
            app.UseRouting();
            app.UseAuthorization();
            app.UseEndpoints(endpoints =>
            {
// ...
```

**Reply by npalla** [reply]
It's not the proxy routing that throwing 404, it's the UI request. Is there any way we can do the same fix before the Request starting ...
![image](https://user-images.githubusercontent.com/3278465/108229792-1e7e3300-710e-11eb-99f6-0b22ed833fcb.png)

**Reply by Tratcher** [reply]
Those logs look like you've applied the above workaround? Where? Doing it at the top of Startup.Configure should have been early enough.
